### PR TITLE
RATIS-1420. Improve the appendEntity with inconsistency entities case

### PR DESCRIPTION
**Pull request by ChenSammi**
https://issues.apache.org/jira/browse/RATIS-1420

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java [hunk 1]
```diff
@@ -1297,22 +1297,17 @@ private long checkInconsistentAppendEntries(TermIndex previous, LogEntryProto...
...
-
       final long commitIndex =  state.getLog().getLastCommittedIndex();
-      if (commitIndex > 0 && commitIndex >= firstEntryIndex) {
-        LOG.info("{}: Failed appendEntries: the first entry (index {}) is already committed (commit index: {})",
-            getMemberId(), firstEntryIndex, commitIndex);
-        return commitIndex + 1;
+      final long nextIndex = Math.max(snapshotIndex, commitIndex);
+      if (nextIndex > 0 && nextIndex >= firstEntryIndex) {
+        LOG.info("{}: Failed appendEntries as the first entry (index {}) is already existed (index: {})",
+            getMemberId(), firstEntryIndex, nextIndex);
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:1309]: Let's print out both the snapshotIndex and the commitIndex.
```
        LOG.info("{}: Failed appendEntries as the first entry (index {})" +
                " already exists (snapshotIndex: {}, commitIndex: {})",
            getMemberId(), firstEntryIndex, snapshotIndex, commitIndex);
```

**Review by szetszwo — COMMENTED** [review]
The change looks good.  Just a minor comment on the log message.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java (hunk 1 above)
Comment (ChenSammi) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java:1309]: Sure.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by ChenSammi** [comment]
Thanks @szetszwo  for the code review.

**Pull request merged**